Services/WatchlistService.swift

```diff
@@ -52,28 +54,38 @@ private class Container {
     }
 }
 
-class WatchlistService: WatchlistServiceProtocol {
+typealias WatchlistService = CoreDataWatchlistService
+//typealias WatchlistService = UserDefaultsWatchlistService
+
+class CoreDataWatchlistService: WatchlistServiceProtocol {
     private let container = Container.shared
 
-    func getStatus(for id: Int) throws -> WatchlistStatus {
-        let request = WatchlistMovie.createFetchRequest()
+    func getStatus(for id: Int, completion: (Result<WatchlistStatus, WatchlistServiceError>)-> Void) {
+        let request = CoreDataWatchlistItem.createFetchRequest()
         let predicate = NSPredicate(format: "id == %d", id)
         request.predicate = predicate
         
         do {
             let movies = try container.viewContext.count(for: request)
             if movies != 0 {
-                return .added
+                completion(.success(.added))
             } else {
-                return .notAdded
+                completion(.success(.notAdded))
             }
         } catch {
-            throw WatchlistServiceError.failedToFetchFromPersistentStore
+//            I'm not sure about the form of recording errors. This is what I've come up with so far. Tried recodring just WatchlistServiceError.failedToFetchFromPersistentStore. Looked ok to me as well, however NSError allows for sending additional info.
```

Comment:
Another way here is to add a computed property to your error:
```swift
enum MyError: Int, Error {
    case failedToDoSomething
    
    var asNSError: NSError {
        NSError(domain: "MyErrorDomain", code: rawValue, userInfo: nil)
    }
}
```
and use here as:
```swift
Crashlytics.crashlytics().record(error: MyError.failedToDoSomething.asNSError)
```